daisy/step_wait_for_instances_signal.go

```diff
@@ -178,6 +192,61 @@ func waitForSerialOutput(s *Step, project, zone, name string, so *SerialOutput,
 	}
 }
 
+func waitForGuestAttribute(s *Step, project, zone, name string, ga *GuestAttribute, interval time.Duration) DError {
+	w := s.w
+	msg := fmt.Sprintf("Instance %q: watching for key %s", name, ga.KeyName)
+	if ga.SuccessValue != "" {
+		msg += fmt.Sprintf(", SuccessValue: %q", ga.SuccessValue)
+	}
+	w.LogStepInfo(s.name, "WaitForInstancesSignal", msg+".")
+	var errs int
+	tick := time.Tick(interval)
+	for {
+		select {
+		case <-s.w.Cancel:
+			return nil
+		case <-tick:
+			resp, err := w.ComputeClient.GetGuestAttributes(project, zone, name, "", ga.KeyName)
+			if err != nil {
+				if apiErr, ok := err.(*googleapi.Error); ok && apiErr.Code == 404 {
```

Comment:
Unsure if it's that big a deal but this does potentially get us into an odd spot if the instance gets deleted, we have no way to fast fail in that scenario, ideally we would check of the existence of the instance but since 404 is expected we don't really want to double our compute calls. Not sure what the best solution is here

Reply:
i'm investigating how we can get all attributes back rather than relying on the 404 but i think it may require a change to the golang package for compute API. let's accept this tradeoff for now, as it has to be opted in to